internal/fetchers/oci/manifest/manifest.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"io"
+	"path"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+
+	"github.com/enterprise-contract/ec-cli/internal/fetchers/oci"
+)
+
+func ImageManifests(ctx context.Context, ref name.Reference, opts ...remote.Option) (map[string]json.RawMessage, error) {
+	img, err := oci.NewClient(ctx).Image(ref, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	content := mutate.Extract(img)
+	defer content.Close()
+	archive := tar.NewReader(content)
+
+	manifests := map[string]json.RawMessage{}
+	for {
+		header, err := archive.Next()
```

Comment:
It would be nice if we could use `range` to iterate over this. I don't think it's possible though.

Reply:
Not seeing a way to do it